DuckDuckGo.Fluent.Plugin/DuckDuckGoSearchApp.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Blast.API.Core.Processes;
+using Blast.API.Processes;
+using Blast.Core.Interfaces;
+using Blast.Core.Objects;
+using Blast.Core.Results;
+using TextCopy;
+using static DuckDuckGo.Fluent.Plugin.DuckDuckGoSearchResult;
+using static DuckDuckGo.Fluent.Plugin.JsonResult;
+using static DuckDuckGo.Fluent.Plugin.AppFunctions;
+
+namespace DuckDuckGo.Fluent.Plugin
+{
+    internal class DuckDuckGoSearchApp : ISearchApplication
+    {
+        private const string SearchAppName = "DuckDuckGo Instant Answers";
+        public const string UserAgentString = "Mozilla/5.0 (Windows NT 10.0; Win64; x64)";
+        public static readonly JsonSerializerOptions SerializerOptions = new() {PropertyNameCaseInsensitive = true};
+        private readonly SearchApplicationInfo _applicationInfo;
+        private BitmapImageResult _logoImage;
+
+        public DuckDuckGoSearchApp()
+        {
+            _applicationInfo = new SearchApplicationInfo(SearchAppName,
+                DuckTagDescription, SearchOperations)
+            {
+                MinimumSearchLength = 1,
+                IsProcessSearchEnabled = false,
+                IsProcessSearchOffline = false,
+                SearchTagOnly = true,
+                ApplicationIconGlyph = SearchResultIcon,
+                SearchAllTime = ApplicationSearchTime.Fast,
+                DefaultSearchTags = SearchTags
+            };
+        }
+
+        public SearchApplicationInfo GetApplicationInfo()
+        {
+            return _applicationInfo;
+        }
+
+        public async IAsyncEnumerable<ISearchResult> SearchAsync(SearchRequest searchRequest,
+            [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            string searchedTag = searchRequest.SearchedTag;
+            string searchedText = searchRequest.SearchedText;
+
+            if (!VerifySearchedTerms(ref searchedText, ref searchedTag)) yield break;
+
+            if (searchedTag.Equals(QrTag))
+            {
+                DuckDuckGoSearchResult duckGoSearchResult = await GetQrImage(searchedText);
+                if (duckGoSearchResult != null)
+                    yield return duckGoSearchResult;
+            }
+            else
+            {
+                string url = GetEndpointUrl(searchedText);
+                var channel = Channel.CreateUnbounded<DuckResult>();
```

Comment:
Seems like nothing happens in parallel here or am I missing something? I don't see reason why to use channels if the code logic is synchronous

Reply:
Oops, thanks for pointing that. There used to be an image grabbing code in here, but later removed as API wasn't providing images properly/consistently.